pallets/subtensor/src/staking/set_children.rs

```diff
@@ -130,6 +130,11 @@ impl<T: Config> Pallet<T> {
         // --- 7.1. Insert my new children + proportion list into the map.
         ChildKeys::<T>::insert(hotkey.clone(), netuid, children.clone());
 
+        if children.is_empty() {
+            // If there are no children, remove the ChildkeyTake value
+            ChildkeyTake::<T>::remove(hotkey.clone(), netuid);
```

Comment:
We should not remove `hotkey` childket take here: `hotkey` is a parent. If this parent is someone's child, even if it has no children, it should keep childkey take if it exists.

Reply:
On second thoughts , I dont think we should be clearing these values out regardless , as everyone has the ability to set a childkey take , regardless of whether or not a parent is allocating to them.